it.polito.elite.dog.drivers.modbus.network/src/it/polito/elite/dog/drivers/modbus/network/ModbusDriverImpl.java

```diff
@@ -726,34 +730,42 @@ public void addDriver(ModbusRegisterInfo register, ModbusDriverInstance driver)
 			registers = new HashSet<ModbusRegisterInfo>();
 			this.gatewayAddress2Registers.put(register.getGatewayIPAddress(), registers);
 		}
-		
+
 		// handle the modbus connection
 		if (!this.connectionPool.containsKey(gwAddress))
 		{
 			// open the modbus connection
-			this.openConnection(gwAddress, gwPort, gwProtocolVariant);
-			
-			// check if a poller is already available or not
-			ModbusPoller poller = this.pollerPool.get(gwAddress);
-			
-			// if no poller is currently handling the gateway, then create a new one
-			if (poller == null)
-			{
-				//create a new poller
-				poller = new ModbusPoller(this, gwAddress);
-				
-				// add the thread to the poller pool
-				this.pollerPool.put(gwAddress,poller);
-				
-				// start polling
-				poller.start();
+			switch(gwProtocolVariant) {
+			case TCP:
+			case RTU_TCP:
+			case RTU_UDP:
+				this.openConnection(gwAddress, gwPort, gwProtocolVariant);
+				break;
+			case RTU:
+				this.openConnection(gwAddress, gwPort, serialParameters);
```

Comment:
Why this shall be done in a different method which is 95% equal to the other one?